### PR TITLE
:sparkles: Text input button (EPI-168)

### PR DESCRIPTION
**Pull request by AubaneNourry**
This pull request introduces a new `UITextInputButton` component and enhances the `UIButton` component to support customizable text and background colors. The key changes are as follows:

### New Component Addition:
* Added a new `UITextInputButton` component, which allows for text input functionality. This includes the implementation of the component itself, its metadata, and event handling for key presses and mouse interactions. (`src/graphics/components/UITextInputButton.cpp`, `src/graphics/components/UITextInputButton.hpp`) [[1]](diffhunk://#diff-00bd83b1606092162f3b39037de0b9147310dff31b0c8b686fa286b64ed9f2f6R1-R219) [[2]](diffhunk://#diff-a7b9553aa38688540b2f1d106cef4b15b30ba4deb670ef960250684fa0228085R1-R118)

### Enhancements to Existing Components:
* Updated the `UIButton` component to support customizable text and background colors through new `ColorField` fields. This includes changes to constructors, rendering logic, and metadata. (`src/graphics/components/UIButton.cpp`, `src/graphics/components/UIButton.hpp`) [[1]](diffhunk://#diff-9f5ed8417d742271631712e00031dc1d9ffc4c135ec390ecea96f88574c4bcacR16) [[2]](diffhunk://#diff-9f5ed8417d742271631712e00031dc1d9ffc4c135ec390ecea96f88574c4bcacL26-R32) [[3]](diffhunk://#diff-9f5ed8417d742271631712e00031dc1d9ffc4c135ec390ecea96f88574c4bcacL41-R51) [[4]](diffhunk://#diff-9f5ed8417d742271631712e00031dc1d9ffc4c135ec390ecea96f88574c4bcacL60-R65) [[5]](diffhunk://#diff-9f5ed8417d742271631712e00031dc1d9ffc4c135ec390ecea96f88574c4bcacR98-R112) [[6]](diffhunk://#diff-55aff90d57716a2f2a91001f942dadc2b0f3cfdcf5918f2cef274a784a186869L25-R25) [[7]](diffhunk://#diff-55aff90d57716a2f2a91001f942dadc2b0f3cfdcf5918f2cef274a784a186869R52-R53) [[8]](diffhunk://#diff-55aff90d57716a2f2a91001f942dadc2b0f3cfdcf5918f2cef274a784a186869L156-R165)

### Build Configuration:
* Updated the CMake configuration to include the new `UITextInputButton` component in the build process. (`src/graphics/components/CMakeLists.txt`) [[1]](diffhunk://#diff-3506e95843b9d41586418d2170145d079e36f2f52c33965b05320bb462ef4dabR11) [[2]](diffhunk://#diff-3506e95843b9d41586418d2170145d079e36f2f52c33965b05320bb462ef4dabR20)

### Minor Fixes:
* Corrected a typo in the `UIText` component's metadata description. (`src/graphics/components/UIText.cpp`)

### Component Registration:
* Registered the new `UITextInputButton` component within the engine. (`src/engine/Engine.cpp`) [[1]](diffhunk://#diff-792e7ee9541fb28c61cdf063f032057c64c788f41fe955bae6b5c1a702f2f15bR40) [[2]](diffhunk://#diff-792e7ee9541fb28c61cdf063f032057c64c788f41fe955bae6b5c1a702f2f15bR54)

### CODE REVIEW

File: src/graphics/components/UITextInputButton.cpp [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        if (_capsLock && key.length() == 1 && (std::isalpha(key[0]) != 0)) {
+          _label += static_cast<char>(std::toupper(key[0]));
+        } else {
+          _label += key;
+        }
+      }
+    }
+      _text.setString(_label);
+  }
+}
```
Comment (Marius-P1) [inline src/graphics/components/UITextInputButton.cpp:219]:
```suggestion
}

```

File: src/graphics/components/UITextInputButton.cpp [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+void UITextInputButton::toggleWriting(const EventData &eventData) {
+   _writing = inButton(static_cast<sf::Event::MouseButtonEvent *>(eventData.data));
+    std::cout << "writing: " << _writing << std::endl;
+}
+
+void UITextInputButton::listenToKeys(const EventData &eventData) {
+  if (_writing) {
+      std::cout << "writing" << std::endl;
```
Comment (Marius-P1) [inline src/graphics/components/UITextInputButton.cpp:183]: I think you can remove your debug or you can use the LOG system

File: src/graphics/components/UITextInputButton.cpp [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+      if (_firstInput) {
+          std::cout << "first input" << std::endl;
```
Comment (Marius-P1) [inline src/graphics/components/UITextInputButton.cpp:185]: Here  too

**Review by Marius-P1 — APPROVED** [review]
Nothing to say except for your debug :ok_hand:

**Review by RenardFute — APPROVED** [review]

**Review by Marius-P1 — APPROVED** [review]
:ok_hand:

**Pull request merged**